### DISCUSSION TITLE
Winmerge Installtion path from Windows Registry

### DISCUSSION THREAD

**Discussion by sweetpineapplepie** [WinMerge/winmerge#2207] [Q&A]
Hello guys,

I am looking to find the installation path of Winmerge from the Windows Registry.

Is it possible to get the installation path of Winmerge automatically if it is installed on the PC? 

Example: from Mitsubishi software GXWorks3, I can get the installation path using:

The GX Works3 install folder can be read from the Windows registry:
- 32-bit OS: HKEY_LOCAL_MACHINE\SOFTWARE\MITSUBISHI\SWnDN-GPPW3\CurrentVersion\InstallPath
- 64-bit OS: HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\MITSUBISHI\SWnDN-GPPW3\CurrentVersion\InstallPath

I have seen "\HKEY_CURRENT_USER\SOFTWARE\Thingamahoochie\WinMerge" has parameter "Executable" with value "C:\Program Files\WinMerge\WinMergeU.exe"

Does it always have the same installation path?

**Comment by sdottaka** [comment]
`Executable` in `HKEY_CURRENT_USER\SOFTWARE\Thingamahoochie\WinMerge` is set when you install with the installer or when you press the OK button in the Options window.
If WinMerge exists in a location different from the installation folder, when you run that WinMerge and press the OK button in the Options window, `Executable` will be changed to the path of that WinMerge.

When installed using the installer, the installation path is also stored in the following registry key.

`HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Uninstall\WinMerge_is1` or `InstallLocation` of `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall\WinMerge_is1` (for per-user installation)
